CONTRIBUTING.md

```diff
@@ -29,3 +29,138 @@ Recommendations and requirements for how to best contribute to Parquet. We striv
 ### License
 By contributing your code, you agree to license your contribution under the terms of the APLv2:
 https://github.com/apache/parquet-format/blob/master/LICENSE
+
+### Additions/Changes to the Format
+
+Note: This section applies to actual functional changes to the specification.
+Fixing typos, grammar, and clarifying concepts that would not change the
+semantics of the specification can be done as long a comitter feels comfortable
+to merge them. When in doubt starting a discussion on the dev mailing list is
+encouraged.
+
+The general steps for adding features to the format are as follows:
+
+1. Discuss changes on the developer mailing list (dev@parquet.apache.org).
+   Often times it is helpful to link to a draft pull request to make the
+   discussion concrete. This step is complete when there is lazy consensus. Part
+   of the consensus is whether it sufficient to provide 2 working
+   implementations as outlined in step 2 or if demonstration of the feature with
+   a down-stream query engine is necessary to justify the feature (e.g.
```

Comment:
This is overly specific IMHO. Not all features are developed in relation with a "down-stream query engine". For example, the BYTE_STREAM_SPLIT additions were motivated by size measurements with different encodings and compression algorithms.

Reply:
Right, I think a lot of features will not need this per @etseidl point on a previous revision, this helps avoid extraneous features that sound good in practice but don't provide much value in practice.  Other areas where this is more likely to apply could be for more complicated indexing schemes.